src/main/java/org/generationcp/middleware/service/impl/study/SampleServiceImpl.java

```diff
@@ -225,10 +215,17 @@ public Boolean studyHasSamples(final Integer studyId) {
 	}
 
 	@Override
-	public Boolean studyEntryHasSamples(Integer studyId, Integer entryId) {
+	public Boolean studyEntryHasSamples(final Integer studyId, final Integer entryId) {
 		return this.daoFactory.getSampleDao().studyEntryHasSamples(studyId, entryId);
 	}
 
+	void populateTakenBy(final List<SampleDTO> sampleDTOS) {
+		// Populate takenBy with full name of user from workbench database.
+		final List<Integer> userIds = sampleDTOS.stream().map(sampleDTO -> sampleDTO.getTakenByUserId()).collect(Collectors.toList());
+		final Map<Integer, String> userIDFullNameMap = this.userService.getUserIDFullNameMap(userIds);
+		sampleDTOS.forEach(sampleDTO -> sampleDTO.setTakenBy(userIDFullNameMap.get(sampleDTO.getTakenByUserId())));
+	}
+	
 	protected void setWorkbenchDataManager(final WorkbenchDataManager workbenchDataManager) {
```

Comment:
Existing in master but seems we can remove `workbenchDataManager` from this class